lib/pghero/methods/tables.rb

```diff
@@ -1,67 +1,294 @@
+            )
+            SELECT
+              sum(heap_blks_hit) / nullif(sum(heap_blks_hit) + sum(heap_blks_read), 0) AS rate
+            FROM
+              (
+                SELECT
+                  heap_blks_hit,
+                  heap_blks_read
+                FROM
+                  pg_statio_user_tables
+                UNION ALL
+                SELECT
+                  (json_array_elements(result)->>'heap_blks_hit')::bigint AS heap_blks_hit,
+                  (json_array_elements(result)->>'heap_blks_read')::bigint AS heap_blks_read
+                FROM
+                  worker_heap_blks_stats_json
+              ) AS cluster_heap_blks_stats
+          SQL
+          )
+        end
       end
 
       def table_caching
-        select_all <<-SQL
-          SELECT
-            schemaname AS schema,
-            relname AS table,
-            CASE WHEN heap_blks_hit + heap_blks_read = 0 THEN
-              0
-            ELSE
-              ROUND(1.0 * heap_blks_hit / (heap_blks_hit + heap_blks_read), 2)
-            END AS hit_rate
-          FROM
-            pg_statio_user_tables
-          ORDER BY
-            2 DESC, 1
-        SQL
+        if !citus_enabled?
+          select_all <<-SQL
+            SELECT
+              schemaname AS schema,
+              relname AS table,
+              CASE WHEN heap_blks_hit + heap_blks_read = 0 THEN
+                0
+              ELSE
+                ROUND(1.0 * heap_blks_hit / (heap_blks_hit + heap_blks_read), 2)
+              END AS hit_rate
+            FROM
+              pg_statio_user_tables
+            ORDER BY
+              2 DESC, 1
+          SQL
+        else
+          select_all <<-SQL
+            WITH dist_heap_blks_stats AS (
+              WITH placements_heap_blks_stats AS (
+                SELECT
+                  logicalrelid,
+                  (run_command_on_placements(logicalrelid, $$
+                    SELECT
+                      heap_blks_hit AS hit
+                    FROM
+                      pg_statio_user_tables
+                    WHERE
+                      relid = '%s'::regclass::oid $$ )).result::bigint AS heap_blks_hit,
+                  (run_command_on_placements(logicalrelid, $$
+                    SELECT
+                      heap_blks_read
+                    FROM
+                      pg_statio_user_tables
+                    WHERE
+                      relid = '%s'::regclass::oid $$ )).result::bigint AS heap_blks_read
+                FROM
+                  pg_dist_partition
+              )
+              SELECT
+                n.nspname AS schemaname,
+                c.relname,
+                sum(p.heap_blks_hit) AS heap_blks_hit,
+                sum(p.heap_blks_read) AS heap_blks_read
+              FROM
+                placements_heap_blks_stats p
+                JOIN pg_class c ON p.logicalrelid::oid = c.oid
+                LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
```

Comment:
Why `LEFT JOIN`?

Reply:
Just following the same pattern
https://github.com/ankane/pghero/blob/819a4e596256f0b7463e3f6be2bdd7fcd8eae8a8/lib/pghero/methods/space.rb#L17
https://github.com/ankane/pghero/blob/819a4e596256f0b7463e3f6be2bdd7fcd8eae8a8/lib/pghero/methods/space.rb#L37